shared/keystore/keystore.go

```diff
@@ -0,0 +1,37 @@
+package keystore
+
+import (
+	"crypto/ecdsa"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"io/ioutil"
+	"os"
+)
+
+func RetrieveJson(directory string, password string) ([]byte, error) {
+	f, err := os.Open(directory)
+	if err != nil {
+		return nil, err
+	}
+
+	keyjson, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+	return keyjson, nil
+}
+
+func DecryptKeystore(keyjson []byte, password string) (ecdsa.PublicKey, error) {
+	key, err := keystore.DecryptKey(keyjson, password)
+	if err != nil {
+		return ecdsa.PublicKey{}, err
```

Comment:
note: we won't be using ecdsa but rather bls, so feel free to use the `bls.PublicKey{}` definition from the `shared/bls` package here